### DISCUSSION TITLE
github actions/gh pages error - ERROR: The specified path (_site) does not exist so cannot be published.

### DISCUSSION THREAD

**Discussion by DOH-FAA3303** [quarto-dev/quarto-cli#5147] [Q&A]
I wasn't able to push changes to my gh-pages due to this error, however once I removed `_site` from my .gitignore I could successfully push the changes and the document rendered via github actions.

Has this changed in the last few months? I used to have `_site` in my gitignore and the site rendered just fine. Do we now need to push `_site` files? It's still in the quarto documentation that I can have _site in the gitignore https://quarto.org/docs/publishing/github-pages.html#ignoring-output

**Comment by mcanouil** [comment]
Could you share the repository where you are having this issue?

**Comment by cderv** [comment]
Also check the Github configuration for your repository https://quarto.org/docs/publishing/github-pages.html#source-branch
Is your website correctly built from `gh-pages` ?

**Comment by DOH-FAA3303** [comment]
@mcanouil unfortunately it is a private repo:/ But if there is something in specific you'd like to see I can send a screenshot.

@cderv - yes I have it built from gh-pages. It's strange, everything has worked smoothly since I made the page ~ 6 months ago but just yesterday it required the `_site` folder to be added in the commit. I added the _site folder in a commit and then put it back in the .gitignore and everything seems to be working again. Strange.. but I can close the discussion!
![image](https://user-images.githubusercontent.com/92396451/231230924-a04ba240-c362-417e-9b6d-ed60f2e81ae1.png)

**Reply by cderv** [reply]
Oh strange. Glad it works. Maybe a github change somehow... 🤷

**Reply by mcanouil** [reply]
I believe there is something wrong either in the repository or in the GitHub repository.

I'll probably try later to reproduce this somehow, meanwhile I would suggest to delete the `gh-pages` branch, to remove the generated site and generated files from the main branch such as `_site`.
Check that you are pointing in your GitHub Actions workflows to the up to date version of Quarto action.
Use Quarto CLI (I would suggest v1.3 pre-release) to publish your site to GitHub Pages to réinitialise the branch properly.

**Reply by cderv** [reply]
> everything seems to be working again. Strange.. but I can close the discussion!

I think everything is working fine already @mcanouil, from what I can read in the reply. Do you still think something is off ? 
Asking in case we need to update our doc.

**Reply by mcanouil** [reply]
@cderv the following statement which "solve" the issue is a bit off as this should not have changed anything.

> yesterday it required the _site folder to be added in the commit. I added the _site folder in a commit and then put it back in the .gitignore and everything seems to be working again.

**Reply by cderv** [reply]
Ok if you think there is something to dig to improve our recommendation, that is good then. 
I understood this was ok and could be a temporary side effect of some past change (like https://github.blog/2022-08-10-github-pages-now-uses-actions-by-default/ almost 6 months ago). And as discussion closed by @DOH-FAA3303 I felt everything was ok.

**Comment by mcanouil** [comment]
@DOH-FAA3303 I did the complete setup for GitHub Pages and it worked just fine without having to add `_site` at any point.
See <https://github.com/mcanouil/quarto-gh-pages>.

**Reply by DOH-FAA3303** [reply]
I'll try what you mentioned above - removing gh-pages branch, the site, and re-running the process. I'm sure it was just a bug on my end! Thanks for the help though!

**Reply by mcanouil** [reply]
The Quarto GitHub Actions and Quarto CLI evolved a lot in the last few months, so it could also be that. ;)
